File: frontend/src/lib/api/blogs.ts
import { api } from "./client";
import type { BlogPost } from "@/types";

export const blogsApi = {
  list: (domainId?: string, status?: string) => {
    const params = new URLSearchParams();
    if (domainId) params.set("domain_id", domainId);
    if (status) params.set("status", status);
    return api.get<BlogPost[]>(`/api/v1/blogs?${params}`);
  },
  get: (id: string) => api.get<BlogPost>(`/api/v1/blogs/${id}`),
  generate: (data: {
    domain_id: string;
    keyword_id?: string;
    custom_keyword?: string;
    custom_title?: string;
    word_count?: number;
  }) => api.post<BlogPost>("/api/v1/blogs/generate", data),
  update: (id: string, data: { title?: string; meta_description?: string; body_html?: string }) =>
    api.put<BlogPost>(`/api/v1/blogs/${id}`, data),
  publish: (id: string, data: { wp_url?: string; wp_username?: string; wp_password?: string; publish_status?: string }) =>
    api.post<BlogPost>(`/api/v1/blogs/${id}/publish`, data),
};

File: backend/app/api/schemas/blogs.py
from pydantic import BaseModel


class BlogGenerateRequest(BaseModel):
    domain_id: str
    keyword_id: str | None = None
    custom_keyword: str | None = None
    custom_title: str | None = None
    word_count: int = 1500


class BlogUpdateRequest(BaseModel):
    title: str | None = None
    meta_description: str | None = None
    body_html: str | None = None


class BlogPublishRequest(BaseModel):
    wp_url: str | None = None
    wp_username: str | None = None
    wp_password: str | None = None
    publish_status: str = "draft"


class BlogResponse(BaseModel):
    id: str
    domain_id: str
    title: str
    slug: str
    meta_description: str | None = None
    body_html: str | None = None
    target_keyword: str | None = None
    search_volume: int | None = None
    difficulty: str | None = None
    word_count: int | None = None
    status: str = "draft"
    wp_post_id: int | None = None
    wp_url: str | None = None
    created_at: str | None = None
